### PR TITLE
Refactor `syncService.start` method

### PR DESCRIPTION
**Pull request by AnkushinDaniil**
- Extracted logic for getting the next block height into a separate method `getNextHeight`.
- Moved block processing logic into a new method `processBlock` to reduce the complexity of the `start` method.
- Improved error handling by using `fmt.Errorf` to wrap errors.
- Enhanced code readability and maintainability by breaking down large functions into smaller.

These changes aim to reduce complexity.

### CODE REVIEW

**Review by kirugan — APPROVED** [review]

File: p2p/sync.go [hunk 1]
```diff
@@ -60,83 +59,86 @@ func (s *syncService) start(ctx context.Context) {
...
 		iterCtx, cancelIteration := context.WithCancel(ctx)
-
-		var nextHeight int
-		if curHeight, err := s.blockchain.Height(); err == nil {
-			nextHeight = int(curHeight) + 1
-		} else if !errors.Is(err, db.ErrKeyNotFound) {
-			s.log.Errorw("Failed to get current height", "err", err)
+		nextHeight, err := s.getNextHeight()
+		if err != nil {
+			s.logError("Failed to get current height", err)
```
Comment (weiihann) [inline p2p/sync.go:64]: I'd suggest we add block height data in the error log

File: p2p/sync.go [hunk 2]
```diff
@@ -60,83 +59,86 @@ func (s *syncService) start(ctx context.Context) {
...
 
 		s.log.Infow("Start Pipeline", "Current height", nextHeight-1, "Start", nextHeight)
 
 		// todo change iteration to fetch several objects uint64(min(blockBehind, maxBlocks))
 		blockNumber := uint64(nextHeight)
-		headersAndSigsCh, err := s.genHeadersAndSigs(iterCtx, blockNumber)
-		if err != nil {
-			s.logError("Failed to get block headers parts", err)
+		if err := s.processBlock(iterCtx, blockNumber); err != nil {
+			s.logError("Failed to process block", err)
```
Comment (weiihann) [inline p2p/sync.go:74]: Suggest to add block number in the error log

File: p2p/sync.go [hunk 3]
```diff
@@ -60,83 +59,86 @@ func (s *syncService) start(ctx context.Context) {
...
 		iterCtx, cancelIteration := context.WithCancel(ctx)
-
-		var nextHeight int
-		if curHeight, err := s.blockchain.Height(); err == nil {
-			nextHeight = int(curHeight) + 1
-		} else if !errors.Is(err, db.ErrKeyNotFound) {
-			s.log.Errorw("Failed to get current height", "err", err)
+		nextHeight, err := s.getNextHeight()
+		if err != nil {
+			s.logError("Failed to get current height", fmt.Errorf("blockNumber: %d, err: %w", nextHeight, err))
```
Comment (kirugan) [inline p2p/sync.go:64]: in case of error returned value is always zero

File: p2p/sync.go (hunk 1 above)
Comment (derrix060) [reply p2p/sync.go:64]:
```suggestion
			s.logError("Failed to get current height ", fmt.Errorf("current Height: %d, err: %w", nextHeight-1, err)
```

**Review by weiihann — APPROVED** [review]

**Pull request merged**